includes/class-duplicate-payment-prevention-service.php

```diff
@@ -226,4 +228,42 @@ public function clear_session_processing_order_after_landing_order_received_page
 			$this->remove_session_processing_order( $order_id );
 		}
 	}
+
+	/**
+	 * Remove and cancel attached intent ID to the order if exists.
+	 *
+	 * @param  WC_Order $order Order object.
+	 *
+	 * @return void
+	 */
+	public function remove_and_cancel_attached_intent_id( WC_Order $order ): void {
+		$intent_id_attached_to_order = $this->order_service->get_intent_id_for_order( $order );
+		if ( ! empty( $intent_id_attached_to_order ) ) {
+			try {
+				$request = Cancel_Intention::create( $intent_id_attached_to_order );
+				$request->send( 'wcpay_cancel_intent_request', $order );
```

Comment:
I think we changed the signature here. Look into `apply_hook` and `set_hook_args`